gridstatus/ercot.py

```diff
@@ -491,6 +845,73 @@ def _handle_html_data(self, df, columns):
         cols_to_keep = ["Time"] + list(columns.keys())
         return df[cols_to_keep].rename(columns=columns)
 
+    def _filter_by_settlement_point_type(self, df, location_type):
+        """Filter by settlement point type"""
+        norm_location_type = location_type.upper()
+        if norm_location_type == LOCATION_TYPE_NODE:
+            df = df[df["SettlementPointType"].isin(RESOURCE_NODE_SETTLEMENT_TYPES)]
+        elif norm_location_type == LOCATION_TYPE_ZONE:
+            df = df[df["SettlementPointType"].isin(LOAD_ZONE_SETTLEMENT_TYPES)]
+        elif norm_location_type == LOCATION_TYPE_HUB:
+            df = df[df["SettlementPointType"].isin(HUB_SETTLEMENT_TYPES)]
+        else:
+            raise ValueError(f"Invalid location_type: {location_type}")
+        return df
+
+    def _filter_by_locations(self, df, field_name, locations):
+        """Filter settlement point name by locations list"""
+        if isinstance(locations, list):
+            df = df[df[field_name].isin(locations)]
+        return df
+
+    def _get_location_type_name(self, location_type):
+        norm_location_type = location_type.upper()
+        if norm_location_type == LOCATION_TYPE_NODE:
+            return "Node"
+        elif norm_location_type == LOCATION_TYPE_ZONE:
+            return "Zone"
+        elif norm_location_type == LOCATION_TYPE_HUB:
+            return "Hub"
+        else:
+            raise ValueError(f"Invalid location_type: {location_type}")
+
+    def _get_settlement_point_mapping(self, verbose=False):
+        """Get dataframe whose columns can help us filter out values"""
+
+        report_type_id = SETTLEMENT_POINTS_LIST_AND_ELECTRICAL_BUSES_MAPPING_RTID
+        url = f"https://www.ercot.com/misapp/servlets/IceDocListJsonWS?reportTypeId={report_type_id}"
+        if verbose:
+            print(f"Fetching document {url}", file=sys.stderr)
+        docs = self._get_json(url)["ListDocsByRptTypeRes"]["DocumentList"]
```

Comment:
any reason you can use `_get_document` for this?

Reply:
can you use `_get_document` to get the url, but then still inline the zip file extraction? 

basically, this logic seems to already be in `_get_document`

```python
report_type_id = SETTLEMENT_POINTS_LIST_AND_ELECTRICAL_BUSES_MAPPING_RTID
url = f"https://www.ercot.com/misapp/servlets/IceDocListJsonWS?reportTypeId={report_type_id}"
if verbose:
    print(f"Fetching document {url}", file=sys.stderr)
docs = self._get_json(url)["ListDocsByRptTypeRes"]["DocumentList"]
latest_doc = sorted(
    docs, key=lambda x: x["Document"]["PublishDate"])[-1]
doc_id = latest_doc["Document"]["DocID"]
doc_url = f"https://www.ercot.com/misdownload/servlets/mirDownload?doclookupId={doc_id}"
```

Reply:
Yep, I see what you mean now. I was misreading `_get_document` and thinking it wouldn't match anything without a `date`, but I was wrong, and there will be a new `_get_document` anyhow.